engine/src/main/java/io/camunda/zeebe/engine/processing/bpmn/behavior/BpmnEventPublicationBehavior.java

```diff
@@ -79,4 +86,50 @@ public Either<Failure, CatchEventTuple> findErrorCatchEvent(
     final var flowScopeInstance = elementInstanceState.getInstance(context.getFlowScopeKey());
     return catchEventAnalyzer.findCatchEvent(errorCode, flowScopeInstance, Optional.empty());
   }
+
+  public Optional<CatchEventTuple> findEscalationCatchEvent(
+      final DirectBuffer escalationCode, final BpmnElementContext context) {
+    final var flowScopeInstance = elementInstanceState.getInstance(context.getFlowScopeKey());
+    return catchEventAnalyzer.findEscalationCatchEvent(escalationCode, flowScopeInstance);
+  }
+
+  /**
+   * Throws an escalation event to the given element instance/catch event pair. Only throws the
+   * event if the given element instance is exists and is accepting events, e.g. isn't terminating,
+   * wasn't interrupted, etc.
+   *
+   * @param record the record of the escalation
+   * @param escalationCatchEvent a tuple representing a catch event and its current instance
+   * @return returns true if the escalation throw event can be completed, false otherwise
+   */
+  public boolean throwEscalationEvent(
+      final EscalationRecord record, final Optional<CatchEventTuple> escalationCatchEvent) {
```

Comment:
❌ Generally it's a bad practice to pass an `Optional` as a parameter. If there is no catch event present we should not call this method at all and handle this on the `IntermediateThrowEventProcessor`.

Reply:
I just saw I did the same thing in the `CatchEventAnalyzer` a year ago 🙈 Let's try to not introduce more of these cases though 😄

Reply:
If you apply the change underneath this one the optional should already be gone from the parameters.